src/core/graph.ts

```diff
@@ -144,23 +42,32 @@ export class DependencyGraph<T> {
     }
   }
 
+
+  /**
+   * Add the given node to this graph.
+   * @param node 
+   */
   addNode(node: T): void {
     if (!this.adjacencyMap.has(node)) {
       this.adjacencyMap.set(node, new Set());
     }
   }
-
-  getEdges(node: T): Set<T> {
-    // FIXME: use different terminology: origins/effects
-    const nodes = this.adjacencyMap.get(node);
-    if (nodes !== undefined) {
-      return nodes;
-    } else {
-      return new Set<T>();
-    }
+  
+  /**
+   * Return the set of all incoming nodes of the given node.
+   * Incoming nodes are immediate upstream neighbors.
+   * @param node The node to retrieve the incoming nodes of.
+   */
+  getInNodes(node: T): Set<T> {
```

Comment:
Personally I still prefer `upstream` and `downstream`, because "in node/"in vertex" doesn't seem like a terminology that is very commonly used in terms of graphs. `parent` and `children` works as well

Reply:
The problem with `upstream` and `downstream` fails to specify how far up or downstream. I think the definitions given in the comments are pretty clear, no? Incoming nodes are immediate upstream neighbors. Outgoing nodes are immediate downstream neighbors.

Reply:
> The problem is that upstream and downstream fail to specify how far up or downstream
I agree, that might be an issue

> I think the definitions given in the comments are pretty clear, no?
Yes, yet the issue is people might need to read the comments to know what exactly it is. Typically "incoming" and "outgoing" are adjectives to "edges" (e.g. "incoming edges"); wrt this use I think the term parent/ancestors and children/descendants are more commonly used and easily understandable.

But since these methods are well-documented I think people will know what they are.